apps/nextjs-app-router/src/app/[lang]/[[...path]]/page.tsx

```diff
@@ -26,10 +25,11 @@ export async function generateStaticParams() {
   ])
 }
 
-export default async function Page({ params }: { params: ParsedUrlQuery }) {
+export default async function Page(props: { params: ParsedUrlQuery }) {
+  const params = await props.params
   const path = '/' + (params?.path ?? []).join('/')
   const snapshot = await client.getPageSnapshot(path, {
-    siteVersion: getSiteVersion(),
+    siteVersion: await getSiteVersion(),
```

Comment:
Breaking `runtime` change: `getSiteVersion` is now async, as it depends on `draftMode` and `cookies`: https://github.com/makeswift/makeswift/pull/857/files#diff-a84011079bd98e3597f4073734c142606a88c2267d6478719139d3ba6fb505d0R29

Reply:
Is the expectation that for Next.js 14 we still expect users to `await getSiteVersion()`?

Reply:
For the App Router and this new `runtime` release, yes.

Reply:
So is this new runtime release not compatible with older versions of Next.js? Or perhaps it's compatible, but it would require users to change their code even if they're still using an older version (e.g., Next.js 14)?

Reply:
The latter: it's compatible, but it'd require this change. We can allow `siteVersion` to be a `Promise` here if we want to keep the Next.js 14 upgrade path as smooth as possible.

Reply:
Updated `getPageSnapshot` to allow `siteVersion` to be a `Promise` so this can literally be just a Next.js 15 compatibility release with no breaking changes.